visivo/query/runner.py

```diff
@@ -60,73 +63,94 @@ def run(self):
                     job.set_future(executor.submit(job.action, **job.kwargs))
                     job.future.add_done_callback(self.job_callback)
 
-        if len(self.errors) > 0 and self.soft_failure:
+        if len(self.failed_job_results) > 0 and self.soft_failure:
             Logger.instance().error(
-                f"\nRefresh failed in {round(time()-start_time, 2)}s with {len(self.errors)} query error(s)."
+                f"\nRefresh failed in {round(time()-start_time, 2)}s with {len(self.failed_job_results)} query error(s)."
             )
-        elif len(self.errors) > 0 and not self.soft_failure:
+        elif len(self.failed_job_results) > 0 and not self.soft_failure:
             Logger.instance().error(
-                f"\nRun failed in {round(time()-start_time, 2)}s with {len(self.errors)} query error(s)"
+                f"\nRun failed in {round(time()-start_time, 2)}s with {len(self.failed_job_results)} query error(s)"
             )
             exit(1)
         else:
             Logger.instance().info(f"\nRun finished in {round(time()-start_time, 2)}s")
 
-    def update_job_queue(self, job_tracker: JobTracker) -> bool:
-        all_dependencies_completed = True
-        for job in self.jobs:
-            job_item_children = all_descendants(dag=self.dag, from_node=job.item)
-            dependencies = list(
-                filter(
-                    lambda j: job != j and job_item_children.has_node(j.item),
-                    self.jobs,
-                )
-            )
-            incomplete_dependencies = list(
-                filter(
-                    lambda d: not job_tracker.is_job_name_done(d.name),
-                    dependencies,
-                )
-            )
-            dependencies_completed = len(incomplete_dependencies) == 0
-            if not dependencies_completed:
-                all_dependencies_completed = False
+    def job_callback(self, future: Future):
+        job_result: JobResult = future.result(timeout=0.01)
+        if job_result.success:
+            Logger.instance().success(str(job_result.message))
+            self.successful_job_results.append(job_result)
+        else:
+            Logger.instance().error(str(job_result.message))
+            self.failed_job_results.append(job_result)
+
+    def create_job_dag(self):
```

Comment:
Per our conversation with Dan, we can leverage this dag by doing two smaller pieces of work:
1. Add a command to generate a dag: `visivo dag --jobs`.  This command would print a standard output of this or the project dag.
2. Allow you to run a singular job: `visivo run --job job_name`.  This command would bypass our dag and simply run the job for that item.

What can that do you ask?  On mint:
1. Run the dag command and then use dynamic tasks to generate a tree of tasks that duplicate our dag.  https://www.rwx.com/docs/mint/dynamic-tasks
2. Mint becomes a dag runner for Visivo